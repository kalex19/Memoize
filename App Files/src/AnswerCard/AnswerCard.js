import React, {Component} from 'react';
import Style from './AnswerCard.scss';


export default class AnswerCard extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  //on change - toggle checked
  // checkRadio() {
  //   if(this button checked){
  //     uncheck other two
  //   } else if (this button checked){
  //     uncheck other two
  //   } else (this button checked) {
  //     uncheck other two
  //   }
  // }

  // submitAnswer(e, props) {
  //   console.log(this.props)
  //   e.preventDefault()
  //   if(e.target.id === this.props.answerId){
  //     this.addScore()
  //     this.switchAnswer()
  //   } else {
  //     this.noScoreAnimation()
  //     this.switchAnswer()
  //   }
  //   console.log('e', e.target.id);
  // }

  // addScore(props) {
  //   this.setState({
  //    score: this.state.score++
  //   })
      //this.scoreAnimation()
  // }

  // scoreAnimation(){
  //   add class to score box that illuminates the score box when score is given
  //   how to pass this up to App?
  // }

  // noScoreAnimation() {
  //   fly through animation
  // }

  // switchAnswer() {
  //  this.setState({
  //   questionSetkey: questionSetKey++
  //  })
  //should rerender and switch to the next question
  //need to have condition is questionSetKey = 30, trigger another function to end game & reset (endGame)
  // }

  //endGame() {
    //pass up to app
    //replace submit button with restart button which resets state, clears inputs, refreshes page
  //}


render() {
  let answerOne = this.props.answer[0];
  let answerTwo = this.props.answer[1];
  let answerThree = this.props.answer[2];

  return (
    <section className="AnswerCard-container">
      <div className="Answer-card">
      <input id="One" type="radio" className="Button-one"/>
      <span>{answerOne}</span>
      </div>
      <div className="Answer-card">
      <input id="two" type="radio" className="Button-two" />
      <span>{answerTwo}</span>
      </div>
      <div className="Answer-card">
      <input id="three" type="radio" className="Button-three" />
      <span>{answerThree}</span>
      </div>
        <button className="Answer-submit" onClick={this.submitAnswer}>Submit</button>
    </section>
    )
}

} 