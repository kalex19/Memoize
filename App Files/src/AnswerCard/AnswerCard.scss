@import '../variables.scss';

.AnswerCard-container {
  font-size: 1.2em;
}

.Answer-instructions {
  color: $main-color; 
  font-size: 1.5em;
}

.Answer-card {
  color: $main-color; 
  font-size: 1em;
  margin: 2vh 1vh;
}

//Radio Buttons

.Button-one,
.Button-two,
.Button-three {
  margin: 1vh;
}

//Submit Button

.Answer-submit {
  color: $main-color;
  background-color: $background-color;
  border: $button-accent-color 3px dashed;
  font-size: 2rem;
  margin: 3vh;
  padding: 1vh;
}

//Media Queries

@media screen and (max-width: 414px) {
  .AnswerCard-container {
    margin: 0 5%;
    width: 90%;
  }
}
