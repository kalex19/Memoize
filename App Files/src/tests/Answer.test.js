import React from 'react';
import AnswerCard from '../AnswerCard/AnswerCard.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

    const mockDataObj=   {
        "questionSetKey": 1,
        "language": "HTML",
        "codeSnippetQ": "https://imgur.com/9AC4cch",
        "answer": ["className should be class", "className should be class and the regexp[a - Z]\\w+ should be[A - Z]\\w+", "The label should be in a seperate div from the input div"],
        "answerId": 2
      };
    const mockScore=0;
    const mockQuestionSetKey=1;
    const mockScoreAnimation =jest.fn();
    const mockNoScoreAnimation=jest.fn();
    const mockIncrementKey=jest.fn();
    const mockIncrementScore=jest.fn();
    const mockSubmitAnswer=jest.fn();
    const mockRadioState=jest.fn();
    const mockSwitchAnswer=jest.fn();
    const event = { preventDefault: () => {} };

describe('AnswerCard', () => {
    let wrapper;
  
    beforeEach(() => {
      wrapper = shallow(
        <AnswerCard
        dataObj={mockDataObj}
        score={mockScore}
        questionSetKey={mockQuestionSetKey}
        switchAnswer={mockSwitchAnswer}
        scoreAnimation ={mockScoreAnimation}
        noScoreAnimation={mockNoScoreAnimation}
        incrementKey={mockIncrementKey}
        incrementScore={mockIncrementScore}
        submitAnswer={mockSubmitAnswer}
        />
      );
    });

    it('should match the snapshot with all data passed in', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have the proper default state', () => {
        expect(wrapper.state()).toEqual({checkedAnswer: 0, noScoreAnimation: false});
    });

    it('should update the checkedAnswer state when radio button is clicked', () => {
        expect(wrapper.state('checkedAnswer')).toEqual(0)
        wrapper.find('.btn').simulate('change', {event:{target:"value"}})
        expect(wrapper.state('checkedAnswer')).toEqual("value");
    });

    it('should submit an answer and verify on submit', () => {
        wrapper.find('.form').simulate('submit', event);
        expect(mockSubmitAnswer).toHaveBeenCalled();
    });

    it('should invoke three methods on click', () => {
        wrapper.find('.form').simulate('submit', event, {state:{ checkedAnswer: mockDataObj.answerId}});
        expect(mockIncrementScore).toHaveBeenCalled();
        expect(mockScoreAnimation).toHaveBeenCalled();
        expect(mockSwitchAnswer).toHaveBeenCalled();
        expect(mockNoScoreAnimation).not.toBeCalled();
    });
  });
