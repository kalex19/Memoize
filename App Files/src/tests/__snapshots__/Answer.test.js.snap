// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerCard should match the snapshot with all data passed in 1`] = `
<section
  className="AnswerCard-container"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="Answer-card"
    >
      <input
        checked={true}
        className="Button-one btn"
        id="answerOne"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        htmlFor="answerOne"
      >
        className should be class
      </label>
    </div>
    <div
      className="Answer-card"
    >
      <input
        className="Button-two btn"
        id="answerTwo"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        htmlFor="answerTwo"
      >
        className should be class and the regexp[a - Z]\\w+ should be[A - Z]\\w+
      </label>
    </div>
    <div
      className="Answer-card"
    >
      <input
        className="Button-three btn"
        id="answerThree"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        htmlFor="answerThree"
      >
        The label should be in a seperate div from the input div
      </label>
    </div>
    <input
      className="Answer-submit"
      type="submit"
    />
  </form>
</section>
`;
